server/views/components/promo/index.njk

```diff
@@ -1,14 +1,64 @@
-<div class="promo">
-  <img class="promo__image"
-    src="{{ image.sources[(image.sources | length) - 1].src }}"
-      srcset="{%- for source in image.sources %}
-        {{ source.src }} {{ source.size }}w
-        {%- if (source !== (image.sources | last)) %}, {%- endif %}
-      {%- endfor %}"
-      alt="{{ image.alt }}" />
+{% set gallery = 'actions/gallery-link' %}
+{% set podcast = 'actions/volume' %}
+{% set video = 'actions/play' %}
+{#TODO full width variant?#}
+{#TODO full width/dark background variant?#}
+{#TODO Test with screen reader?#}
+{#TODO styling video/podcast/gallery version#}
+<a href="{{ url }}" class="promo  {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}promo--{{meta.type}}{% endif %} {% for modifier in modifiers %}promo--{{ modifier }} {% endfor %} ">
+  <div class="promo__image-container">
+    <img class="promo__image"
+      src="{{ image.sources[(image.sources | length) - 1].src }}"
+        srcset="{%- for source in image.sources %}
+          {{ source.src }} {{ source.size }}w
+          {%- if (source !== (image.sources | last)) %}, {%- endif %}
+        {%- endfor %}"
+        alt="" />
+
+    {% for modifier in modifiers %}
+      {% if modifier == 'series' %}
+        <svg class="promo__defs">{# TODO proper filter - see Tomi #}
+          <filter id="duotone" color-interpolation-filters="sRGB" x="0" y="0" height="100%" width="100%">
+            <feColorMatrix type="matrix"
+              values="0.90 0 0 0   0.40 
+                      0.95 0 0 0  -0.10  
+                    -0.20 0 0 0   0.65 
+                        0  0 0 1   0" />
+          </filter>
+        </svg>
+        <img class="promo__image-mask promo__clip-path--{{ [1,2,3,4,5,6,7,8,9,10] | random }}" src="{{image.sources[(image.sources | length) - 1].src}}" alt="" />
+      {% endif %}
+    {% endfor %}
+
+    {% if meta.type == 'gallery' or meta.type == 'podcast' or meta.type == 'video' %}
+      <div class="promo__icon-container">
+        {% if meta.type == 'gallery' %}
+          {% icon gallery %}
+        {% elif meta.type == 'podcast' %}
+          {% icon podcast %}
+        {% elif meta.type == 'video' %}
+            {% icon video %}
+        {% endif %}
+        {% if meta.type == 'gallery' and meta.length %}
+          <span class="promo__length"><span class="promo__sr">Gallery with</span> {{ meta.length }} <span class="promo__sr">items</span></span>
```

Comment:
In the admittedly unlikely event of having a gallery with one item, would `item{{ 's' if meta.length !== 1 }}` be safer? (not sure if that expression works).